src/config_store.rs

```diff
@@ -121,8 +134,39 @@ impl ConfigStore {
         }
     }
 
-    /// Drop an entire partition of check configurations.
-    pub fn drop_partition(&mut self, partition: u16) {
+    /// Get all check configs across all partitions and buckets.
+    pub fn all_configs(&self) -> Vec<Arc<CheckConfig>> {
+        self.configs.values().cloned().collect()
+    }
+
+    /// Get all check configs that are scheduled for a given tick.
+    pub fn get_configs(&self, tick: Tick) -> TickBucket {
+        self.partitioned_buckets
+            .values()
+            .flat_map(move |b| b[tick.index].iter().cloned())
+            .collect()
+    }
+
+    /// Update the set of partitions that the ConfigStore is responsible for.
+    pub fn update_partitions(&mut self, new_partitions: &HashSet<u16>) {
+        let known_partitions: HashSet<_> = self.partitioned_buckets.keys().cloned().collect();
+
+        // Drop partitions that we are no longer responsible for
+        for removed_part in known_partitions.difference(new_partitions) {
+            self.drop_partition(*removed_part);
+        }
+
+        // Add new partitions
+        for new_partiton in new_partitions.difference(&known_partitions) {
+            self.partitioned_buckets
+                .entry(*new_partiton)
+                .or_insert_with(make_empty_tick_buckets);
+        }
+    }
+
+    /// Drop an entire partition of check configurations. This does NOT update the known set of
+    /// partitions. Use update_partitions.
+    fn drop_partition(&mut self, partition: u16) {
```

Comment:
Should this be private so that no one external can call it?

Reply:
It is, you need `pub` to make it public.